src/OrchardCore.Modules/OrchardCore.Secrets/Services/RSAGenerator.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+namespace OrchardCore.Secrets.Services;
+
+public static class RSAGenerator
+{
+    public static RSA GenerateRSASecurityKey(int size)
+    {
+        // By default, the default RSA implementation used by .NET Core relies on the newest Windows CNG APIs.
```

Comment:
FYI, things have changed in .NET 8.0: instead of returning an implementation based on Windows NCrypt (that relies on the privileged Windows LSASS process to delegate all the crypto operations and store the keys out-of-process), the new implementation now returned by default by `RSA.Create()` uses Windows BCrypt, that only supports in-memory crypto operations: while it's certainly less secure, it's also more efficient (which is why the BCL team made the change). I haven't tested, but I also suspect it could address the issue this helper was trying to fix.

I don't know what the plans regarding older TFMs in OrchardCore vNext are, but if only .NET 8.0 is supported, it may be worth giving it a try and see if we can remove that 😄

Reply:
We only target .NET 8 now, yep.

Reply:
Okay thanks for the info, as soon as it can be validated we could remove this.